### PR TITLE
ember-cli addon

### PR DESCRIPTION
**Pull request by tchak**
I am publishing the PR so there could be more eyes on it. It is not ready to merge.

- [ ] cleanup generated files
- [ ] complete the README with new instructions
- [ ] make sure I did not make some stuff worse in the process of refactoring :)
- [ ] add some basic tests

Mainly I would like some feedback on how I manage configuration. Is it too much magic?

### CODE REVIEW

**Comment by tchak** [comment]
@nolanlawson 
MIT licence was generated by `ember-addon` generator :) This is part of "cleanup generated files". I have no opinion at all on this.
I will update `README` with more instructions.

**Comment by nolanlawson** [comment]
No prob. I'll merge when you're ready, then add a Travis config for tests per @stefanpenner's instructions. Seems like everything is very tidy with Ember Data. :)

File: addon/adapters/pouch.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  defaultSerializer: '-pouch',
+  coalesceFindRequests: true,
+  schema: null,
+  url: null,
+
+  setup: function(url) {
+    url = url || this.url;
+
+    if (url) { this.db = new PouchDB(url); }
+  }.on('init'),
```
Comment (devinrhode2) [inline addon/adapters/pouch.js]: @stefanpenner has anyone considered detecting if super was called or not, and then intelligently calling it for the user when necessary? I just noticed I had some js library code that does this exact thing

Comment (stefanpenner) [inline addon/adapters/pouch.js]: Unfortunately that defeats the point of `super` as it is already an optional call. Subclassing is a refinement on the ancestor classes, when refining a methods it is the choice of the author to fully replace, or merely refine parts of the inherited methods functionality. Super prevents the need to always fully re-implement when refining.

**Comment by nolanlawson** [comment]
@tchak What's the status of this? BTW I added you to the contributors. :)

**Comment by broerse** [comment]
Can we get this ready for production? What is the main problem?

**Comment by nolanlawson** [comment]
It looks like somebody already wrote one: https://github.com/mattmarcum/ember-cli-pouch

@mattmarcum can you comment? Should this be merged into ember-pouch itself?

**Comment by fivetanley** [comment]
@tchak @mattmarcum would love to get this going forward <#

**Comment by fsmanuel** [comment]
@tchak i can take over your branch if you don't have time to finish it. please let me know.

**Comment by nolanlawson** [comment]
@fsmanuel I'd say go ahead. Whoever writes a working Ember CLI integration first, I'll merge that. ;)

And if you could write Travis tests too, that would be icing on the cake. :D

**Comment by fsmanuel** [comment]
@nolanlawson the ember-cli addon comes with tests and a travis.yml. I'll set up ember-try so we can test against different ember versions. I guess I'll start next monday.

**Comment by tchak** [comment]
@fsmanuel awesome. Sorry I lost momentum on this. Glad you taking it over.

**Comment by nolanlawson** [comment]
#55 beat you to the punch! Closing this issue, thanks for starting discussion though. :)

**Pull request closed without merging**